### PR TITLE
[TypeDeclaration] Remove ArrowFunction from ReturnTypeFromReturnDirectArrayRector and ReturnTypeFromReturnNewRector

### PR DESCRIPTION
**Pull request by samsonasik**
Part of resolve this issue effort:

- https://github.com/rectorphp/rector/issues/8678 

Ref from PR comment:

- https://github.com/rectorphp/rector-src/pull/6029#issuecomment-2188731185

For `ArrowFunction`, there is already rule that cover return type on `ArrowFunction`

- `AddArrowFunctionReturnTypeRector` 

so no need to duplicate process in the following rules:

- `ReturnTypeFromReturnDirectArrayRector`
- `ReturnTypeFromReturnNewRector`

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules-tests/TypeDeclaration/Rector/ArrowFunction/AddArrowFunctionReturnTypeRector/Fixture/return_array.php.inc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\TypeDeclaration\Rector\ArrowFunction\AddArrowFunctionReturnTypeRector\Fixture;
```
Comment (TomasVotruba) [inline rules-tests/TypeDeclaration/Rector/ArrowFunction/AddArrowFunctionReturnTypeRector/Fixture/return_array.php.inc:5]: The "Add*Rector" naming is used for configurable rules to add spefic known types, e.g. during Symfony upgrade, but this rule is not configurable.

Name like `ArrowFunctionReturnTypeFrom*Rector` would be better.

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ArrowFunction/AddArrowFunctionReturnTypeRector/Fixture/return_array.php.inc:5]: There is existing `AddArrowFunctionReturnTypeRector` rule , it is moving the fixture as functionality covered already in `AddArrowFunctionReturnTypeRector`

Comment (TomasVotruba) [reply rules-tests/TypeDeclaration/Rector/ArrowFunction/AddArrowFunctionReturnTypeRector/Fixture/return_array.php.inc:5]: I see :+1:

**Comment by TomasVotruba** [comment]
Thanks :clap:

**Pull request merged**